### DISCUSSION TITLE
IPv6 records missing from FoundryVTT internet services

### DISCUSSION THREAD

**Discussion by felddy** [felddy/foundryvtt-docker#752] [General]
The internet services that support FoundryVTT are no longer publishing `AAAA` records to allow IPv6 access.   This was noted in issue #531.  I'm starting this discussion to track any changes.  Services that need support include:

## Status ##

| Name  | `A` | `AAAA` | Notes |
|------|:-:|:-----:|-------|
| api.foundryvtt.com | ✅ | ❌ |  |
| foundryvtt.com | ✅ | ❌ |  |
| foundryvtt.s3.amazonaws.com | ✅ | ❌ | `CNAME` to s3-us-west-2-w.amazonaws.com. |
| s3-us-west-2-w.amazonaws.com. | ✅ | ❌ |  |
| foundryvtt.s3.dualstack.us-west-2.amazonaws.com | ✅ | ✅ | `CNAME` to s3-r-w.dualstack.us-west-2.amazonaws.com. |
| s3-r-w.dualstack.us-west-2.amazonaws.com. | ✅ | ✅ |  |

Regenerate the markdown table above with this gist:
- https://gist.github.com/felddy/8021c23a97a5c269e3b0f0a9b25b9cf6

## References ##

Github issues:
- https://github.com/foundryvtt/foundryvtt/issues/9897
- https://github.com/felddy/foundryvtt-docker/issues/531

Discord:
- `@daegalus`: https://discord.com/channels/170995199584108546/978002307251863552/1127121221063028806
```markdown
Hey, anyone know any info on why `foundryvtt.com` no longer has `AAAA` records for IPv6? I run Foundry on a IPv6 only vm and it can't access the site to pull builds and such from my account because it can't resolve the domain. https://www.nslookup.io/domains/foundryvtt.com/dns-records/
```

**Comment by daegalus** [comment]
So, without knowing how they setup their cloud resources, my guess is they would need to do the following to enable IPv6

* Enable IPv6 on the VPC where the servers live
* Enable IPv6 on any load balancers they use after enabling the VPC ones.
* If using Cloudfront, make sure IPv6 is appropriately enabled.

I haven't worked in AWS for a bit, but thats roughly what I know about how to enable it on their end.

If it is all enabled, it could be as simple as adding the IPv6 address of the load balancer or server to the AAAA record.

The problem is, it is all on their side. We can't do anything but be polite and wait for them to fix it.

**Reply by felddy** [reply]
I've opened an [issue with the main upstream project](https://github.com/foundryvtt/foundryvtt/issues/9897). 

@daegalus, if you have a chance could you post some log output from your Foundry client showing the errors in an IPv6-only environment to that issue?  Not the errors from the tooling of the container, but the client itself not being able to hit the licensing server, buckets, etc...

Thanks!

See:
- https://github.com/foundryvtt/foundryvtt/issues/9897

**Reply by daegalus** [reply]
Ummm, ya, let me set up a new ipv6 only setup. In exasperation, I ended up enabling ipv4 on the box. Might take me a day or 2, just busy.

**Reply by felddy** [reply]
Lol.  I didn't mean to throw you under the bus.  Please don't stress it.  It think the Foundry devs can imagine what the behavior would be.

**Comment by daegalus** [comment]
In the mean time, for those that need a temporary solution without adding IPv4, you can use https://nat64.net/ to get access to ipv4 resources on ipv6 only networks.

I learned of this from this blog post on the state of ipv6: https://matduggan.com/ipv6-is-a-disaster-and-its-our-fault/
